packages/circuits/src/initial.circom

```diff
@@ -0,0 +1,89 @@
+pragma circom 2.0.0;
+
+include "./merkle.circom";
+include "../node_modules/circomlib/circuits/comparators.circom";
+
+template CheckReplaceArray(n) {
+    signal input index;
+    signal input newValue;
+    signal input array[n];
+    signal output newArray[n];
+    signal output replacedValue;
+
+    component checkIndex[n];
+    signal items[n];
+    for (var i = 0; i < n; i++) {
+        checkIndex[i] = IsEqual();
+        checkIndex[i].in[0] <== i;
+        checkIndex[i].in[1] <== index;
+        newArray[i] <== (newValue - array[i]) * checkIndex[i].out + array[i];
```

Comment:
Maybe comment the line above with `newArray[i] = (i == index) ? newValue : array[i]`?

Reply:
Actually I see below that you use `something <== x ? y : z`, can't we use the same pattern here?

I'm thinking maybe not, because below, `x` is only dependent on the loop index, and not on the signal values.

Reply:
Yeap, that's right - basically any conditional statement in Circom cannot uses an unknown value during compilation, otherwise it will run into non quadratic constraint error.